tests/block_device/test_block_device_auth.py

```diff
@@ -0,0 +1,186 @@
+"""Tests for Gen1 BlockDevice Shelly authentication functionality."""
+
+import pytest
+from unittest.mock import AsyncMock, MagicMock
+
+
+def create_mock_block_device():
+    """Create a mock BlockDevice for testing."""
+    from aioshelly.block_device import BlockDevice
+
+    device = MagicMock(spec=BlockDevice)
+    device.shelly = {"auth": False, "type": "SHSW-1", "fw": "20230913-114244/v1.14.0-gcb84623"}
+    device.http_request = AsyncMock(return_value={"enabled": True, "unprotected": False, "username": "admin"})
+    return device
+
+
+class TestBlockDeviceSetAuth:
```

Comment:
We do not create separate classes for test cases, please adapt your tests to the current structure.